data/org.freedesktop.impl.portal.InputCapture.xml

```diff
@@ -0,0 +1,431 @@
+              If the id is missing, the input capture was not triggered by a
+              pointer barrier.
+
+              Where more than one pointer barrier are triggered by the same
+              movement it is up to the compositor to choose one barrier (or use
+              a barrier id of zero).
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <signal name="Activated">
+      <arg type="o" name="session_handle" direction="out"/>
+      <arg type="a{sv}" name="options" direction="out"/>
+    </signal>
+
+    <!--
+        Deactivated:
+        @session_handle: Object path for the #org.freedesktop.impl.portal.Session object representing the session
+        @options: Vardict with optional further information
+
+        The Deactivated signal is emitted when input capture stopped and input events
+        are no longer sent to the application. To prevent future input
+        capture, an application must call org.freedesktop.portal.InputCapture.Disable().
+
+        Supported keys in the @options vardict include:
+        <variablelist>
+          <varlistentry>
+            <term>activation_id u</term>
+            <listitem><para>
+                The same activation_id as in the corresponding
+                #org.freedesktop.impl.portal.InputCapture::Activated signal.
+            </para></listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>cursor_position dd</term>
+            <listitem><para>
+              The current cursor position. Note that this position may be
+              outside the Zones available to this session - this indicates
+              movement larger than a single pixel.
+
+              For example, a fast movement against a barrier on the right edge
+              of a screen may logically put the cursor dozens of pixels into
+              the (non-existing) screen on the other side of the barrier.
+              It is the application's responsibility to adjust the cursor
+              position as necessary.
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <signal name="Deactivated">
+      <arg type="o" name="session_handle" direction="out"/>
+      <arg type="a{sv}" name="options" direction="out"/>
+    </signal>
+
+    <!--
+        ZonesChanged:
+        @session_handle: Object path for the #org.freedesktop.impl.portal.Session object representing the session
+        @options: Vardict with optional further information
+
+        The ZonesChanged signal is emitted when one or more of the zones
+        available **to this session** change. An application should immediately call
+        org.freedesktop.portal.InputCapture.GetZones() to update its state of the zones.
+    -->
+    <signal name="ZonesChanged">
+      <arg type="o" name="session_handle" direction="out"/>
+      <arg type="a{sv}" name="options" direction="out"/>
+    </signal>
```

Comment:
This signal should include the `zone_set` number as used in the `SetPointerBarrier` request, so applications can be sure of which zones are invalidated. Needs a fair bit of spec comments because (for easy of implementation) the zone_set should be associated with zones but if an application never calls `GetZones` it may never see a particular number invalidated later. Plus the theoretical race condition of calling `GetZones` while the signal is already in-flight, so the application gets the new (valid) `zone_set` returned together with the signal for the old one.

The difficult bit here is simply writing the spec :)